backend/src/Repositories/SubmissionRepository.cs

```diff
@@ -0,0 +1,71 @@
+using FormAPI.Data;
+using FormAPI.Models;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.EntityFrameworkCore;
+
+namespace FormAPI.Repositories;
+
+public interface ISubmissionRepository
+{
+    Task<List<SubmissionEntity>> GetAll();
+    Task<SubmissionEntity> GetSingle(Guid submissionId);
+    Task<List<SubmissionEntity>> GetFormSubmissions(Guid formId);
+    Task<SubmissionEntity?> Create(SubmissionEntity entity);
+    Task<SubmissionEntity> Delete(Guid submussionId);
+}
+
+public class SubmissionRepository : ISubmissionRepository
+{
+    private readonly ApiDbContext _context;
+
+    public SubmissionRepository(ApiDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<List<SubmissionEntity>> GetAll()
+    {
+        var submissionList = await _context.Submissions.ToListAsync();
+        return submissionList;
+    }
+
+    public async Task<SubmissionEntity> GetSingle(Guid submissionId)
+    {
+        var entity = await _context.Submissions.FindAsync(submissionId);
+
+        if (entity == null)
+        {
+            Console.WriteLine($"Submission with ID {submissionId} was not found.");
+            return null;
```

Comment:
We have a warning here that the method maybe return null , and we do not allow for that, so we can add `nullable `type `?` to SubmissionEntity in line 32.

Reply:
I did not understand those errors before, but now I do. Thanks for explaining 😄 
Fixed ✅